Mage.Sets/src/mage/cards/p/PantlazaSunFavored.java

```diff
@@ -0,0 +1,94 @@
+package mage.cards.p;
+
+import java.util.UUID;
+import mage.MageInt;
+import mage.abilities.TriggeredAbilityImpl;
+import mage.abilities.effects.keyword.DiscoverEffect;
+import mage.abilities.hint.StaticHint;
+import mage.constants.SubType;
+import mage.constants.SuperType;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.constants.Zone;
+import mage.game.Game;
+import mage.game.events.GameEvent;
+import mage.game.permanent.Permanent;
+
+/**
+ *
+ * @author jimga150
+ */
+public final class PantlazaSunFavored extends CardImpl {
+
+    public PantlazaSunFavored(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{2}{R}{G}{W}");
+        
+        this.supertype.add(SuperType.LEGENDARY);
+        this.subtype.add(SubType.DINOSAUR);
+        this.power = new MageInt(4);
+        this.toughness = new MageInt(4);
+
+        // Whenever Pantlaza, Sun-Favored or another Dinosaur enters the battlefield under your control,
+        // you may discover X, where X is that creature's toughness. Do this only once each turn.
+        this.addAbility(new PantlazaSunFavoredTriggeredAbility());
+    }
+
+    private PantlazaSunFavored(final PantlazaSunFavored card) {
+        super(card);
+    }
+
+    @Override
+    public PantlazaSunFavored copy() {
+        return new PantlazaSunFavored(this);
+    }
+}
+
+class PantlazaSunFavoredTriggeredAbility extends TriggeredAbilityImpl {
```

Comment:
Try using `EntersBattlefieldThisOrAnotherTriggeredAbility` to avoid reimplementing code. (What you have here works in most cases, but would be bugged in the scenario where a permanent ends up with this ability but without having the Dinosaur subtype itself.) There are a few options:
1. Extend that class rather than `TriggeredAbilityImpl`, and add the discover effect in `checkTrigger` after calling the `super`
2. Create a custom effect for "discover X, where X is that creature's toughness" that gets the `permanentEnteringBattlefield` value saved by the trigger, finds the permanent or LKI, and calls `DiscoverEffect.doDiscover` using its toughness. (this is probably what I would do here)
3. currently `DiscoverEffect` doesn't support a `DynamicValue`, but the most maintainable solution would be something akin to `SavedDamageValue` that calculates the toughness and can be used directly in the effect constructor (but this is more refactoring than really needed for this card)

Reply:
Ah, thank you for the help--i'll have to take some time to get to know the convenience ability implementations. I'll give option 1 a shot

Reply:
My solution ended up being a combination of 1 and 2. i'm not 100% sure how i would do purely 1 since  EntersBattlefieldThisOrAnotherTriggeredAbility is looking to get an Effect passed in for its constructor, so i implemented my own effect that gets the toughness of the entering creature and does a discover from that creature's toughness.